src/journal/bsod.c

```diff
@@ -107,7 +109,26 @@ static int acquire_first_emergency_log_message(char **ret) {
         if (!message)
                 return log_oom();
 
-        *ret = TAKE_PTR(message);
+        r = sd_journal_get_data(j, "MESSAGE_ID", &d, &l);
+        if (r == -ENOENT) {
+                *ret_message = TAKE_PTR(message);
+                *ret_message_id = NULL;
+                return 0;
+        }
+        if (r < 0)
+                return log_error_errno(r, "Failed to read message ID from journal: %m");
+
+        d = memory_startswith(d, l, "MESSAGE_ID=");
+        if(!d)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Wrong message ID.");
+
+        assert(l >= STRLEN("MESSAGE_ID="));
```

Comment:
@yuwata , this assertion is already done inside the call ro ```memory_startswith()```
Do we need it here still?

Reply:
this can be dropped